packages/gatsby/src/internal-plugins/query-runner/query-compiler.js

```diff
@@ -62,11 +62,9 @@ const validationRules = [
 let lastRunHadErrors = null
 const overlayErrorID = `graphql-compiler`
 
-const resolveThemes = (plugins = []) =>
-  plugins.reduce((merged, plugin) => {
-    if (plugin.name.includes(`gatsby-theme-`)) {
-      merged.push(plugin.resolve)
-    }
+const resolveThemes = (themes = []) =>
+  themes.reduce((merged, plugin) => {
```

Comment:
It seems like `themes` in redux state is not array,
but it's object that contains subfield `themes` ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/redux/reducers/themes.js )

see https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/gatsby-node.js#L14 for example where `themes.themes.map` is used

I'm not sure we need it to be object like that to be honest, but just to get it going, let's adjust here
```suggestion
  themes.themes.reduce((merged, plugin) => {
```

Reply:
the payload passed into the store is `themes.themes` which should be an array. However, my recent fix addressing this is still unsuccessful.